src/components/CreateRecommendationModal/index.tsx

```diff
@@ -0,0 +1,318 @@
+import { useState, useCallback } from 'react';
+import { Modal, Form, Input, Button, AutoComplete, Upload, Spin, message, DatePicker } from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+import * as Yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm, Controller } from 'react-hook-form';
+import { useQuery } from '@tanstack/react-query';
+import { debounce } from 'lodash';
+import styles from './CreateRecommendationModal.module.scss';
+import { RecommendationService } from '@/api/services/recommendation.service';
+import { MatchService } from '@/api/services/match.service';
+
+// Recommendation Schema with Yup validation
+const RecommendationSchema = Yup.object().shape({
+    title: Yup.string().min(3, 'Title is too short').required('Title is required'),
+    description: Yup.string().min(10, 'Description is too short').required('Description is required'),
+    picture: Yup.string().optional(),
+    country: Yup.string().required('Country is required'),
+    stadium: Yup.string().required('Stadium is required'),
+    league: Yup.string().required('League is required'),
+    homeTeam: Yup.string().required('Home Team is required'),
+    awayTeam: Yup.string().required('Away Team is required'),
+    date: Yup.date().required('Match date is required'),
+});
+
+type CreateRecommendationModalProps = {
+    isOpen: boolean;
+    onClose: () => void;
+};
+
+type CreateRecommendationModalValues = Yup.InferType<typeof RecommendationSchema>;
+
+const API_KEY = 'e859464540085b15535c5ed431595955';
+
+const CreateRecommendationModal = ({ isOpen, onClose }: CreateRecommendationModalProps) => {
+    const [imageUrl, setImageUrl] = useState<string>('');
+    const [uploading, setUploading] = useState(false);
+    const [selectedCountry, setSelectedCountry] = useState('');
+    const [selectedLeague, setSelectedLeague] = useState<number | undefined>();
```

Comment:
```suggestion
    const [selectedLeague, setSelectedLeague] = useState<number>();
```
if empty initial value. type is auto infered as the type passed in generic or `undefined`